nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -76,6 +77,17 @@ in {
       default = "/var/lib/nextcloud";
       description = "Storage path of nextcloud.";
     };
+    datadir = mkOption {
+      type = types.str;
+      default = null;
```

Comment:
If you set it to `null` by default, you have to use `types.nullOr types.str` as `type`. Otherwise you'll get an error like this:

```
error: A definition for option `services.nextcloud.datadir' is not of type `string'. Definition values:
       - In `/home/ma27/Projects/nixpkgs/nixos/modules/services/web-apps/nextcloud.nix': null
(use '--show-trace' to show detailed location information)

```